### ISSUE TITLE
GenericService泛化调用使用oninvoke、onreturn、onthrow时，FutureFilter的invocation.getMethodName()无法正确返回

### ISSUE THREAD

**Issue by xiangxiuhui** [apache/incubator-dubbo#3408]
- [x] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.6.4

### Steps to reproduce this issue

1. GenericService泛化调用
2. 使用oninvoke、onreturn、onthrow
3. FutureFilter无法正常回调

### Expected Result
StaticContext.getKey(invoker.getUrl(), invocation.getMethodName(), Constants.ON_INVOKE_METHOD_KEY)
返回
com.alibaba.boot.dubbo.demo.consumer.DemoService:1.0.0.sayHello.oninvoke.method

### Actual Result
StaticContext.getKey(invoker.getUrl(), invocation.getMethodName(), Constants.ON_INVOKE_METHOD_KEY)
返回
com.alibaba.boot.dubbo.demo.consumer.DemoService:1.0.0.$invoke.oninvoke.method

预期是：

```
        String methodName = invocation.getMethodName();
        if("$invoke".equals(methodName)) {
            Object[] arguments = invocation.getArguments();
            methodName = (String) arguments[0];
        }
```

**Comment by tswstarplanet** [comment]
Can you provider a demo project to reproduce the problem ?

**Comment by tswstarplanet** [comment]
Thank you! I will fix this bug in 2.6.x version

**Comment by chickenlj** [comment]
@tswstarplanet Have you confirmed this issue? Would you please also submit one to 2.7.x?

**Comment by tswstarplanet** [comment]
@chickenlj OK, I will fix it both on 2.6.x and 2.7.x

**Comment by tswstarplanet** [comment]
I debugged it on 2.6.x, it indeedly have problems.
The corresponding code on 2.7.x is refactored. So I will confirm if there are problems, if so, I will fix it.
